app/routes.py

```diff
@@ -0,0 +1,357 @@
+        phone = request_body["phone"]
+    )
+
+    db.session.add(new_customer)
+    db.session.commit()
+
+    return jsonify({"id": new_customer.id}), 201
+
+#Get
+@customers_bp.route("", methods=["GET"])
+def get_customers():
+    customers = Customer.query.all()
+    customers_response = []
+
+    for customer in customers:
+        customers_response.append(
+            {
+            "id" : customer.id,
+            "name" : customer.name,
+            "postal_code" : customer.postal_code,
+            "phone" : customer.phone,
+            "registered_at" : datetime.now()
+            }
+        )
+    return jsonify(customers_response), 200
+
+@customers_bp.route("/<customer_id>", methods=["GET"])
+def get_customer(customer_id):
+    if customer_id.isnumeric() != True:
+        return {"message": "Customer id provided is not a number."}, 400
+    
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    return {
+        "id" : customer.id,
+        "name" : customer.name,
+        "postal_code" : customer.postal_code,
+        "phone" : customer.phone
+    }
+@customers_bp.route("/<customer_id>", methods=["PUT"])
+def update_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    request_body = request.get_json()
+    
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    if "name" not in request_body or "postal_code" not in request_body or "phone" not in request_body:
+        return {
+        "details": "Invalid data"
+        }, 400
+        
+    customer.name = request_body["name"]
+    customer.phone = request_body["phone"]
+    customer.postal_code = request_body["postal_code"]
+    
+    db.session.commit()
+
+    return {
+        "id" : customer.id,
+        "name" : customer.name,
+        "postal_code" : customer.postal_code,
+        "phone" : customer.phone,
+        "registered_at" : datetime.now()
+    }
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_customer(customer_id):
+
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    db.session.delete(customer)
+    db.session.commit()
+
+    return jsonify({"id": customer.id}), 200
+##Rental code
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+@rentals_bp.route("/check-out", methods = ["POST"])
+def rental_check_out():
+    rental_request = request.get_json()
+
+    if "customer_id" not in rental_request or "video_id" not in rental_request:
+        return jsonify(details="Request body must include customer id and video id"), 400
+
+    customer_id = rental_request["customer_id"]
+    video_id = rental_request["video_id"]
+
+    try: 
+        video_id = int(video_id)
+        customer_id = int(customer_id)
+    except:
+        return jsonify(None), 400
```

Comment:
👍 

Nice check for id validity. This is common to overlook (and some other endpoints could still use a similar check). For that reason, it can be really useful to wrap checks like this into helper functions so that rather than using the raw query.get, we use our "smarter" lookup helper.